libRALFit/src/ral_nlls_internal.f90

```diff
@@ -4118,7 +4118,7 @@ Recursive subroutine evaltensor_J(status, n, m, s, J, params)
        real(wp), dimension(*), intent(in)    :: s
        real(wp), dimension(*), intent(out)   :: J
        class( params_base_type ), intent(inout) :: params
-       integer :: ii, jj
```

Comment:
@tyronerees this is the proposal for `evaltensor_J`to address slices inside assumed-shape arrays, can you review it? just to make sure everything is right.

Reply:
This looks good to me